### PR TITLE
feat(connector): add http email connector

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add new email connector to send emails by HTTP API calls.

## Testing

Unit tests and local tested

截屏2024-09-18 15 34 38

截屏2024-09-18 15 34 47

截屏2024-09-18 15 35 54

## Checklist

- [ ] `.changeset` (will be added later)
- [x] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

File: packages/connectors/connector-http-email/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        );
+
+        throw new ConnectorError(ConnectorErrorCodes.General, rawBody);
+      }
+
+      throw new ConnectorError(ConnectorErrorCodes.General, error);
+    }
+  };
+
+const createSendGridMailConnector: CreateConnector<EmailConnector> = async ({ getConfig }) => {
```
Comment (simeng-li) [inline packages/connectors/connector-http-email/src/index.ts:61]: should use another name?

File: packages/connectors/connector-http-email/src/constant.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import type { ConnectorMetadata } from '@logto/connector-kit';
+import { ConnectorConfigFormItemType } from '@logto/connector-kit';
+
+export const defaultMetadata: ConnectorMetadata = {
+  id: 'http-email-service',
+  target: 'http-mail',
+  platform: null,
+  name: {
+    en: 'HTTP Email',
+    'zh-CN': 'HTTP 邮件',
```
Comment (charIeszhao) [inline packages/connectors/connector-http-email/src/constant.ts:10]: No need to do the translation anymore

File: packages/connectors/connector-http-email/src/constant.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  logo: './logo.svg',
+  logoDark: null,
+  description: {
+    en: 'Send email via HTTP call.',
+    'zh-CN': '通过 HTTP 调用发送邮件。',
+  },
```
Comment (charIeszhao) [inline packages/connectors/connector-http-email/src/constant.ts:17]: Same here

File: packages/connectors/connector-http-email/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
+{
+  "name": "@logto/connector-http-email",
+  "version": "1.2.0",
```
Comment (charIeszhao) [inline packages/connectors/connector-http-email/package.json:3]: 0.1.0 maybe

File: packages/connectors/connector-http-email/logo.svg
Comment (charIeszhao) [inline packages/connectors/connector-http-email/logo.svg:1]: Should also have a `logo-dark.svg`

**Review by darcyYe — APPROVED** [review]
Overall LGTM, please resolve other reviewers comments

File: packages/connectors/connector-http-email/logo.svg
Comment (wangsijie) [reply packages/connectors/connector-http-email/logo.svg:1]: added

**Pull request merged**